### PR TITLE
Filter reviews

### PR DESCRIPTION
**Pull request by JoshLove-msft**
image

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -60,6 +60,8 @@
         <script src="~/lib/jquery/dist/jquery.sticky-sidebar.js"></script>
         <script src="~/lib/ResizeSensor/dist/ResizeSensor.js"></script>
         <script src="~/lib/bootstrap/dist/js/bootstrap.bundle.js"></script>
+        <script src="~/lib/mark/dist/jquery.mark.js"></script>
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:63]: Why do we need this?

Comment (JoshLove-msft) [reply src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:63]: this is the dist for dev, just like we have for the other libraries.

Comment (JoshLove-msft) [reply src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:63]: Which I forgot to commit..

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -68,6 +70,9 @@
                 crossorigin="anonymous"
                 integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=">
         </script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js"
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:73]: Or this?

Comment (pakrym) [reply src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:73]: Shouldn't we just have one?

File: src/dotnet/APIView/APIViewWeb/Client/package.json [hunk 3]
```diff
@@ -26,6 +26,7 @@
     "css-loader": "^3.2.0",
     "eslint": "^5.16.0",
     "eslint-plugin-vue": "^5.0.0",
+    "mark.js": "^8.11.1",
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Client/package.json:29]: Can we bundle this into the main JS in webpack?

Comment (JoshLove-msft) [reply src/dotnet/APIView/APIViewWeb/Client/package.json]: Removed

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml (hunk 2 above)
Comment (JoshLove-msft) [reply src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml]: This was supposed to be the jquery one. Updated.

File: src/dotnet/APIView/APIViewWeb/Client/src/reviews.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  // make the search box the initial focused element so users can just start typing once page loads
+  searchBox.focus();
+
+  const context = $(".review-name");
+  searchBox.keyup(function () {
+
+    // highlight matching text using mark.js framework
+    const searchText = (searchBox.val() as string).toUpperCase();
+    (context as any).unmark();
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Client/src/reviews.ts:12]: Can you define variable as any instead?

File: src/dotnet/APIView/APIViewWeb/Client/src/reviews.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    if (searchText) {
+      (context as any).mark(searchText, {
+        done: function () {
+          $(this).not(":has(mark)").hide();
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Client/src/reviews.ts:16]: This is supposed to accomplish `    // hide rows that do not match`

Comment (JoshLove-msft) [reply src/dotnet/APIView/APIViewWeb/Client/src/reviews.ts:16]: No, this is just for the highlighting.

Comment (JoshLove-msft) [reply src/dotnet/APIView/APIViewWeb/Client/src/reviews.ts:16]: Fixed

**Review by pakrym — APPROVED** [review]

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -68,6 +70,10 @@
                 crossorigin="anonymous"
                 integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=">
         </script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js"
+                asp-fallback-src="~/lib/mark/dist/mark.js"
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:74]: min

**Pull request merged**